src/_components/Navbar.jsx

```diff
@@ -1,29 +1,109 @@
 import React from "https://esm.sh/react";
 
-export default function Navbar() {
+// Global navigation links array
+const navigationLinks = [
+  { href: "/mission", text: "Mission" },
+  { href: "/projects", text: "Projects" },
+  { href: "/community", text: "Get Involved" },
+  { href: "/blog", text: "Blog" },
+];
+
+export default function NavigationBar() {
   const logo = "/assets/logos/logo_negative.png";
   return (
-    <nav className="flex items-center justify-around space-x-64 p-6 bg-primary z-50">
-      <a href="/" className="flex items-center text-white mr-60">
-        <img src={logo} alt="Logo" className="mr-2 w-10 h-10" />
-        <span className="self-center text-3xl font-semibold font-poppins whitespace-nowrap dark:text-white">
+    <nav className="relative flex items-center justify-between p-6 bg-primary z-50">
+      {/* Logo and Brand Name Section */}
+      <a href="/" className="flex items-center text-white">
+        <img src={logo} alt="Blueprint Logo" className="mr-2 w-10 h-10" />
+        <span className="text-3xl font-poppins font-semibold hover:font-bold">
           blueprint
         </span>
       </a>
-      <div>
-        <a href="/mission" className="mr-4 text-xl text-white">
-          Mission
-        </a>
-        <a href="/projects" className="mr-4 text-xl text-white">
-          Projects
-        </a>
-        <a href="/community" className="mr-4 text-xl text-white">
-          Get Involved
-        </a>
-        <a href="/blog" className="text-xl text-white">
-          Blog
-        </a>
+      {/* Desktop Navigation Menu - Visible only on large screens */}
+      <div className="hidden lg:flex space-x-6">
+        {renderDesktopNavigationLinks()}
+      </div>
+      {/* Mobile Menu Toggle and Dropdown - Visible on screens smaller than "large" */}
+      <div className="lg:hidden relative">
+        {/* Toggles the mobile menu */}
+        <input
+          type="checkbox"
+          id="mobileMenuToggle"
+          className="hidden peer"
+          aria-label="Toggle mobile menu"
+        />
+        <label htmlFor="mobileMenuToggle" className="block cursor-pointer p-2">
+          {renderHamburgerIcon()}
+        </label>
+
+        {/* Mobile Navigation Menu - Expands and shows when hamburger icon is clicked*/}
+
+        {/* Change background color for dropdwon menu here if desired */}
+        {/* current dropdown menu css:
+        - 16px from top
+        - 2px from right 
+        - bg color = primary */}
+        <div className="absolute top-[calc(100%+1rem)] right-2 bg-primary transition-all duration-300 max-h-0 overflow-hidden peer-checked:max-h-screen peer-checked:border peer-checked:border-white peer-checked:rounded-lg peer-checked:shadow-lg w-48">
+          {renderMobileNavigationLinks()}
+        </div>
       </div>
     </nav>
   );
 }
+
+// Function to render desktop navigation links
+function renderDesktopNavigationLinks() {
+  // Map through the links to render them
```

Comment:
Same comment here on the unnecessary comments

Reply:
Agreed. This is a good example where a well named function doesn't need the extra comments (i.e. your function name is the same as the comment here)

Reply:
Removed